packages/core/src/__tests__/ActiveWorkflows.test.ts

```diff
@@ -0,0 +1,376 @@
+	const workflow = mock<Workflow>();
+	const additionalData = mock<IWorkflowExecuteAdditionalData>();
+	const mode: WorkflowExecuteMode = 'trigger';
+	const activation: WorkflowActivateMode = 'init';
+
+	const getTriggerFunctions = jest.fn() as IGetExecuteTriggerFunctions;
+	const triggerResponse = mock<ITriggerResponse>();
+
+	const pollFunctions = mock<PollContext>();
+	const getPollFunctions = jest.fn<PollContext, unknown[]>();
+
+	LoggerProxy.init(mock());
+	const scheduledTaskManager = mock<ScheduledTaskManager>();
+	const triggersAndPollers = mock<TriggersAndPollers>();
+	const errorReporter = mock<ErrorReporter>();
+	const activeWorkflows = new ActiveWorkflows(
+		scheduledTaskManager,
+		triggersAndPollers,
+		errorReporter,
+	);
+
+	const triggerNode = mock<INode>();
+	const pollNode = mock<INode>();
+	const pollTimes = { item: [{ mode: 'everyMinute' } as TriggerTime] };
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+		getPollFunctions.mockReturnValue(pollFunctions);
+		pollFunctions.getNodeParameter.mockReturnValue(pollTimes);
+		triggersAndPollers.runTrigger.mockResolvedValue(triggerResponse);
+	});
+
+	describe('add()', () => {
+		describe('should activate workflow', () => {
+			it('with trigger nodes', async () => {
+				workflow.getTriggerNodes.mockReturnValue([triggerNode]);
+				workflow.getPollNodes.mockReturnValue([]);
+
+				await activeWorkflows.add(
+					workflowId,
+					workflow,
+					additionalData,
+					mode,
+					activation,
+					getTriggerFunctions,
+					getPollFunctions,
+				);
+
+				expect(activeWorkflows.isActive(workflowId)).toBe(true);
+				expect(workflow.getTriggerNodes).toHaveBeenCalled();
+				expect(triggersAndPollers.runTrigger).toHaveBeenCalledWith(
+					workflow,
+					triggerNode,
+					getTriggerFunctions,
+					additionalData,
+					mode,
+					activation,
+				);
+			});
+
+			it('with polling nodes', async () => {
+				workflow.getTriggerNodes.mockReturnValue([]);
+				workflow.getPollNodes.mockReturnValue([pollNode]);
+
+				await activeWorkflows.add(
+					workflowId,
+					workflow,
+					additionalData,
+					mode,
+					activation,
+					getTriggerFunctions,
+					getPollFunctions,
+				);
+
+				expect(activeWorkflows.isActive(workflowId)).toBe(true);
+				expect(workflow.getPollNodes).toHaveBeenCalled();
+				expect(scheduledTaskManager.registerCron).toHaveBeenCalled();
+			});
+		});
+
+		describe('should throw error', () => {
+			it('if trigger activation fails', async () => {
+				workflow.getTriggerNodes.mockReturnValue([triggerNode]);
+				workflow.getPollNodes.mockReturnValue([]);
+
+				const error = new Error('Trigger activation failed');
+				triggersAndPollers.runTrigger.mockRejectedValueOnce(error);
+
+				await expect(
+					activeWorkflows.add(
+						workflowId,
+						workflow,
+						additionalData,
+						mode,
+						activation,
+						getTriggerFunctions,
+						getPollFunctions,
+					),
+				).rejects.toThrow(WorkflowActivationError);
```

Comment:
Should we check that `isActive` returns false?

Reply:
good catch. adding `expect(activeWorkflows.isActive(workflowId)).toBe(false)` here breaks the test. I think we might have caught a bug here,